### DISCUSSION TITLE
How to call activities based on some conditions

### DISCUSSION THREAD

**Discussion by NiveditaGalwade** [elsa-workflows/elsa-core#5587] [Q&A]
Wanted to apply some condition before calling particular activity in workflow

**Comment by glime-ai** [comment]
To apply a condition before calling a particular activity in an Elsa workflow, you can use the "If" or "Switch" branching activities. These activities allow you to specify conditions and execute different branches of your workflow based on the outcome of those conditions.

Here's how you can use them:

1. **If Activity**: Use the "If" activity to specify a condition. If the condition evaluates to true, the activities in the "then" branch will be executed; otherwise, the activities in the "else" branch will be executed.

2. **Switch Activity**: Use the "Switch" activity when you have multiple conditions to evaluate. You can define cases based on the value of a given expression, and for each case, you can specify different activities to execute.

To use these activities, you would typically place them in your workflow before the activity you want to conditionally execute. Then, you would configure the condition or cases accordingly, and place the activity you want to execute in the appropriate branch of the "If" or "Switch" activity.

For more detailed information on how to use these activities, you can refer to the Elsa documentation on branching activities.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
